packages/aws-cdk/test/api/aws-auth/awscli-compatible.test.ts

```diff
@@ -0,0 +1,26 @@
+import { AwsCliCompatible } from '../../../lib/api/aws-auth/awscli-compatible';
+
+test('does not mess up with session token env variables if they are undefined', async () => {
+  process.env.AWS_ACCESS_KEY_ID = 'foo';
+  process.env.SECRET_ACCESS_KEY = 'bar';
+
+  // Making sure these variables are not defined
+  delete process.env.AWS_SESSION_TOKEN;
+  delete process.env.AMAZON_SESSION_TOKEN;
+
+  await AwsCliCompatible.credentialChainBuilder();
+
+  expect(process.env.AWS_SESSION_TOKEN).toBeUndefined();
+});
+
+test('preserves session token env variables if they are defined', async () => {
```

Comment:
Add a test where `AWS_SESSION_TOKEN` is undefined but `AMAZON_SESSION_TOKEN` is?